src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -92,180 +96,406 @@ struct NetworkInfo
+{
+    CHIP_ERROR encodingError             = CHIP_NO_ERROR;
+    EmberAfNetworkCommissioningError err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_BOUNDS_EXCEEDED;
 
-            using NetworkIDLenType = decltype(sNetworks[i].mNetworkIDLen);
-            VerifyOrExit(chip::CanCastTo<NetworkIDLenType>(ssid.size()), err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_OUT_OF_RANGE);
-            sNetworks[i].mNetworkIDLen = static_cast<NetworkIDLenType>(ssid.size());
+    if (GetClusterState() != ClusterState::kIdle)
+    {
+        err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_UNKNOWN_ERROR;
+    }
+    else
+    {
+        for (size_t i = 0; i < kMaxNetworks; i++)
+        {
+            if (sNetworks[i].mNetworkType == NetworkType::kUndefined)
+            {
+                Thread::OperationalDataset & dataset = sNetworks[i].mData.mThread;
+                CHIP_ERROR error                     = dataset.Init(operationalDataset);
+
+                if (error != CHIP_NO_ERROR)
+                {
+                    ChipLogDetail(Zcl, "Failed to parse Thread operational dataset: %s", ErrorStr(error));
+                    err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_UNKNOWN_ERROR;
+                    break;
+                }
+
+                uint8_t extendedPanId[Thread::kSizeExtendedPanId];
+
+                static_assert(sizeof(sNetworks[i].mNetworkID) >= sizeof(extendedPanId),
+                              "Network ID must be larger than Thread extended PAN ID!");
+                if ((error = dataset.GetExtendedPanId(extendedPanId)) != CHIP_NO_ERROR)
+                {
+                    ChipLogDetail(Zcl, "Failed to get Thread extpanid: %s", ErrorStr(error));
+                    err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_UNKNOWN_ERROR;
+                    break;
+                }
+                memcpy(sNetworks[i].mNetworkID, extendedPanId, sizeof(extendedPanId));
+                sNetworks[i].mNetworkIDLen = sizeof(extendedPanId);
+
+                sNetworks[i].mNetworkType = NetworkType::kThread;
+                sNetworks[i].mEnabled     = false;
+
+                err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_SUCCESS;
+                break;
+            }
+        }
+    }
 
-            sNetworks[i].mNetworkType = NetworkType::kWiFi;
-            sNetworks[i].mEnabled     = false;
+    TLV::TLVWriter * writer          = nullptr;
+    app::CommandPathParams cmdParams = { /* endpoint id */ endpointId, /* group id */ 0, ZCL_NETWORK_COMMISSIONING_CLUSTER_ID,
+                                         ZCL_ADD_THREAD_NETWORK_RESPONSE_COMMAND_ID,
+                                         (chip::app::CommandPathFlags::kEndpointIdValid) };
+    SuccessOrExit(encodingError = apCommandObj->PrepareCommand(&cmdParams));
+    VerifyOrExit((writer = apCommandObj->GetCommandDataElementTLVWriter()) != nullptr, encodingError = CHIP_ERROR_INCORRECT_STATE);
+    SuccessOrExit(encodingError = writer->Put(TLV::ContextTag(0), static_cast<uint32_t>(err)));
```

Comment:
Why is that `static_cast` needed?  There's an overload of `Put` that takes enum values, no?